src/strategies/strategies/uni-v4-telx-lp/index.ts

```diff
@@ -0,0 +1,142 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { getAddress } from '@ethersproject/address';
+import { Multicaller } from '../../utils';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'robriks';
+export const version = '0.1.0';
+
+const REGISTRY_ABI = [
+  'function getAmountsForLiquidity(bytes32 poolId, uint128 liquidity, int24 tickLower, int24 tickUpper) view returns (uint256 amount0, uint256 amount1, uint160 sqrtPriceX96)',
+  'function getSubscriptions(address owner) view returns (uint256[])',
+  'function getPositionDetails(uint256 tokenId) view returns (tuple(address owner, bytes32 poolId, int24 tickLower, int24 tickUpper, uint128 liquidity, tuple(address currency0, address currency1, uint24 fee, int24 tickSpacing, address hooks) poolKey))'
+];
+
+interface TokenInfo {
+  address: string;
+  decimals: number;
+  price: number;
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses, // array of voter addresses to calculate scores for
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const telPrice = options.telPrice;
+  if (!telPrice || telPrice === 0) return {};
+
+  // use generic token price & decimals lookup map for max maintainability
+  const tokenInfoMap = new Map<string, { decimals: number; price: number }>();
+  for (const token of options.tokens as TokenInfo[]) {
```

Comment:
Could we please add a limit for `tokens` to avoid memory problems, ideally throw a error if more tokens are passed